ad9959/src/lib.rs

```diff
@@ -566,6 +612,33 @@ impl ProfileSerializer {
         }
     }
 
+    /// Update the system clock configuration.
+    ///
+    /// # Args
+    /// * `reference_clock_frequency` - The reference clock frequency provided to the AD9959 core.
+    /// * `multiplier` - The frequency multiplier of the system clock. Must be 1 or 4-20.
+    pub fn set_system_clock(
+        &mut self,
+        reference_clock_frequency: f32,
+        multiplier: u8,
+    ) -> Result<f32, Error> {
+        let frequency = reference_clock_frequency * multiplier as f32;
+
+        // The enabled channel will be updated after clock reconfig
+        let mut fr1 = [0u8; 3];
+
+        // The ad9959 crate does not modify FR1[0:17]. These bits keep their default value.
+        // These bits by default are 0.
+        fr1[0].set_bits(2..=6, multiplier);
```

Comment:
What are the other bits of FR1? Can we be sure we won't modify them later in some other portion of the code? I don't like baking assumptions like this into the software. Say definitively why we won't touch the other bits, or just read the register instead.

If we can't read the register, document why and add this to the function limitation docs.